### PR TITLE
feat: add edx-sysadmin plugin app

### PR DESCRIPTION
**Pull request by arslanashraf7**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [ ] Testing
  - [ ] Code is tested
  - [x] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable
  - [ ] Opened issue for DevOps regarding necessary configuration changes to deployed environments

#### What are the relevant tickets?
https://github.com/mitodl/edx-sysadmin/issues/76

**Note 1:**

I have also ported over the tests along with other components. The tests are currently not running because we will need to configure the test runup for the overall open-edx-plugins repo at some point. And these tests would run then.

**Note 2:**
For the initial pass, I ignored most of the warnings with `# noqa:` (No QA) to keep the scope of the PR narrow and try to not break any working functionality. Once we are done with this and everything works smoothly we should try and fix some of the pre-commit errors especially `PTH***` ones.

#### What's this PR do?
- Ports over the https://github.com/mitodl/edx-sysadmin to open-edx-plugins
- Adds the required pants configurations to support package creation of sysadmin
- Adds required plugin settings so that the sysadmin behaves as a plugin within the edX context
- Fixes lots of pre-commit warnings/errors. (There were so many pre-commit issues since this is an old code and was introduced to pre-commit for the first time. I have ignored most of the warnings and fixed the straightforward ones.)

#### How should this be manually tested?
- First of all, install sysadmin as mentioned in https://github.com/mitodl/edx-sysadmin
- Test the features mentioned in https://github.com/mitodl/edx-sysadmin
- Now, Setup up `open-edx-plugins` and generate package build e.g. using `./pants package ::`
- Install the package generated from this repo over the existing installation and see that:
    - There are no duplicate sysadmin packages
    - Everything(User creation, Course deletion, Course addition, Git logs, sysadmin commands such as `git_add_course`) works the same, especially the Git logs and course details.

**Note: (PRs you can take a look at for the steps to test the sysadmin overall, Follow instructions on those PRs)**
- https://github.com/mitodl/edx-sysadmin/pull/31
- https://github.com/mitodl/edx-sysadmin/pull/29
- https://github.com/mitodl/edx-sysadmin/pull/23

#### Where should the reviewer start?
- From an existing installation from the [old repo](https://github.com/mitodl/edx-sysadmin).

#### Screenshots

image

image

image

image

### CODE REVIEW

File: src/edx_sysadmin/requirements/base.in
Comment (blarghmatey) [inline src/edx_sysadmin/requirements/base.in:1]: These requirements should just be added to the top-level pyproject.toml as they won't be used at all. We don't need any of the contents of this requirements directory in the context of this repository.

Comment (arslanashraf7) [reply src/edx_sysadmin/requirements/base.in:1]: Thanks for pointing this out. I was kind of confused on this as well and thought i would decide about keeping or leaving it once the PR is ready for the feedback.

**Review by asadali145 — COMMENTED** [review]
I have tested sysadmin, it works fine. We should update the readme. Also, we should keep the import order. Overall, Looks great. An excellent effort to port the changes.

**Comment by asadali145** [comment]
@blarghmatey This PR is ready for your review.

**Review by blarghmatey — APPROVED** [review]

File: src/edx_sysadmin/api/views.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    get_clean_branch_name,
+    get_local_active_branch,
+    get_local_course_repo,
+)
+from path import Path as get_path  # noqa: N813
+from rest_framework import permissions, status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
```
[the remark this answers is not in the document]
Comment (arslanashraf7) [reply src/edx_sysadmin/api/views.py:22]: That was done by pre-commit, I'll try to change it and see if pre-commit accepts.

File: src/edx_sysadmin/api/urls.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+"""
+URLs for edx_sysadmin.
+"""
+from django.conf.urls import url
+from edx_sysadmin.api.views import (
```
[the remark this answers is not in the document]
Comment (arslanashraf7) [reply src/edx_sysadmin/api/urls.py:5]: That was done by pre-commit, I'll try to change it and see if pre-commit accepts.

File: src/edx_sysadmin/README.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
+edx-sysadmin
```
[the remark this answers is not in the document]
Comment (arslanashraf7) [reply src/edx_sysadmin/README.rst:1]: Good catch, We definitely need to update these.

**Review by asadali145 — CHANGES_REQUESTED** [review]
LGTM

**Review by asadali145 — APPROVED** [review]

**Pull request merged**